src/huggingface_hub/inference/_providers/nscale.py

```diff
@@ -0,0 +1,50 @@
+from typing import Any, Dict, Optional, Union
+import base64
+
+from huggingface_hub.inference._common import RequestParameters, _as_dict
+from ._common import (
+    TaskProviderHelper,
+    BaseConversationalTask,
+    filter_none,
+)
+
+class NscaleTask(TaskProviderHelper):
+    def __init__(self, task: str):
+        super().__init__(provider="nscale", base_url="https://inference.api.nscale.com", task=task)
+
+    def _prepare_route(self, mapped_model: str, api_key: str) -> str:
+        if self.task == "text-to-image":
+            return "/v1/images/generations"
+        elif self.task == "conversational":
+            return "/v1/chat/completions"
+        raise ValueError(f"Unsupported task '{self.task}' for Nscale API.")
+
+class NscaleChatCompletion(BaseConversationalTask):
+    def __init__(self):
+        super().__init__(provider="nscale", base_url="https://inference.api.nscale.com")
+
+class NscaleTextToImageTask(NscaleTask):
+    def __init__(self):
+        super().__init__("text-to-image")
```

Comment:
There is no need for a base `NscaleTask` class and also, let's keep a consistent naming for the classes (i.e. renaming `NscaleChatCompletion` into `NscaleConversationalTask`).

```suggestion
class NscaleConversationalTask(BaseConversationalTask):
    def __init__(self):
        super().__init__(provider="nscale", base_url="https://inference.api.nscale.com")


class NscaleTextToImageTask(TaskProviderHelper):
    def __init__(self):
        super().__init__(provider="nscale", base_url="https://inference.api.nscale.com", task="text-to-image")

    def _prepare_route(self, mapped_model: str, api_key: str) -> str:
        return "/v1/images/generations"

```